cmd/download-plugins/main.go

```diff
@@ -83,21 +83,28 @@ func run() error {
 }
 
 func pluginExistsMatchingDigest(plugin release.PluginRelease, downloadDir string) (bool, error) {
-	digest, err := release.CalculateDigest(filepath.Join(downloadDir, filepath.Base(plugin.URL)))
-	if err != nil {
+	filename := filepath.Join(downloadDir, filepath.Base(plugin.URL))
+	if _, err := os.Stat(filename); err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
 		return false, err
 	}
+	digest, err := release.CalculateDigest(filename)
+	if err != nil {
+		return false, fmt.Errorf("failed to calculate digest for plugin %s: %w", plugin.URL, err)
```

Comment:
Should the error reference the `plugin.URL` or `filename`, which is easier to debug?